src/components/QueryCondition.vue

```diff
@@ -0,0 +1,132 @@
+			:disabled="selectedPropertyValueRelation === propertyValueRelation.Regardless"
+		/>
+	</div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { mapState } from 'vuex';
+import { TextInput } from '@wmde/wikit-vue-components';
+
+import PropertyLookup from '@/components/PropertyLookup.vue';
+import ValueTypeDropDown from '@/components/ValueTypeDropDown.vue';
+import SearchResult from '@/data-access/SearchResult';
+import PropertyValueRelation from '@/data-model/PropertyValueRelation';
+import Error from '@/data-model/Error';
+import allowedDatatypes from '@/allowedDataTypes';
+import Validator from '@/form/Validator';
+
+export default Vue.extend( {
+	name: 'QueryCondition',
+	data() {
+		return {
+			fieldErrors: {
+				property: null as null | Error,
+				value: null as null | Error,
+			},
+			limitedSupport: false,
+			propertyValueRelation: PropertyValueRelation,
+		};
+	},
+	methods: {
+		validate(): void {
+			const formValues = {
+				property: this.selectedProperty,
+				value: this.textInputValue,
+				propertyValueRelation: this.selectedPropertyValueRelation,
+			};
+			const validator = new Validator( formValues );
+			const validationResult = validator.validate();
+			this.fieldErrors = validationResult.fieldErrors;
+			if ( this.selectedProperty !== null ) {
+				this.validateForLimitedSupport( this.selectedProperty );
+			}
+		},
+		validateForLimitedSupport( selectedProperty: SearchResult ): void {
+			this.limitedSupport = false;
+			this.fieldErrors.property = null;
+			if ( selectedProperty && !allowedDatatypes.includes( selectedProperty.datatype ) ) {
+				this.selectedPropertyValueRelation = PropertyValueRelation.Regardless;
+				this.limitedSupport = true;
+				this.fieldErrors.property = {
+					type: 'warning',
+					message: 'query-builder-property-lookup-limited-support-note',
+				};
+			}
+		},
+	},
+	computed: {
+		selectedProperty: {
+			get(): SearchResult | null {
+				return this.$store.getters.property;
+			},
+			set( selectedProperty: SearchResult ): void {
+				this.selectedPropertyValueRelation = PropertyValueRelation.Matching;
+				this.validateForLimitedSupport( selectedProperty );
+				this.$store.dispatch( 'updateProperty', selectedProperty );
+			},
+		},
+		selectedPropertyValueRelation: {
+			get(): PropertyValueRelation {
+				return this.$store.getters.propertyValueRelation;
+			},
+			set( selectedPropertyValueRelation: PropertyValueRelation ): void {
+				if ( selectedPropertyValueRelation === PropertyValueRelation.Regardless ) {
+					this.textInputValue = '';
+				}
+				this.$store.dispatch( 'updatePropertyValueRelation', selectedPropertyValueRelation );
+			},
+		},
+		textInputValue: {
+			get(): string { return this.$store.getters.value; },
+			set( value: string ): void { this.$store.dispatch( 'updateValue', value ); },
+		},
+		...mapState( {
+			errors: 'errors',
+		} ),
+	},
+	watch: {
+		'$store.state.errors': function (): void {
+			this.validate();
```

Comment:
Do we actually need this? What does it do?

Reply:
Yeah we need this. It basically adds the per-field validation messages when the query is incomplete. We should rework it once the state is in a better shape.

Reply:
I see. That works, but is quite magical. Going with the ref after all is maybe the better idea. I.e. having a ref `condition` on the `QueryCondition` in the `QueryBuilder` component and then calling `condition.validate()` in the `QueryBuilder`'s `validate` method.

You may need the following code for giving the correct type to the ref: https://github.com/wmde/wikit/commit/bad24a9eba4a2be4fec9539d973c6636381bd212#diff-c000023c1d09de1efda5d372b6a61658f3158308cbb6d585822d328ac512389eR55

Reply:
Done

Reply:
Yay :)